### PR TITLE
Adding add two numbers

### PR DESCRIPTION
**Pull request by yuri-spizhovyi-mit**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Review by suhrobmuboraksho — COMMENTED** [review]
Thanks for the task. It would make the test more comprehensive if we had tests for edge cases? Thoughts....

**Comment by yuri-spizhovyi-mit** [comment]
Thanks @suhrobmuboraksho. I added descriptions and doctests to add_two_numbers file. Added more tests and description to test file.

File: solutions/tests/test_add_two_numbers.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    def test_non_integer_input(self):
+        """It should raise TypeError for a non-integer input"""
+        with self.assertRaises(TypeError):
+            add_numbers("1", 2)
+
+    def test_one_argument(self):
+        """It should raise TypeError when only one argument is provided"""
+        with self.assertRaises(TypeError):
+            add_numbers(1)
+
```
Comment (suhrobmuboraksho) [inline solutions/tests/test_add_two_numbers.py:74]: Maybe the unittest also needs an edge case where there are no arguments ie add_numbers()

**Review by suhrobmuboraksho — COMMENTED** [review]
Doctest had no mistakes.
Doctype explanation is good.
Comprehensive unittests.
No linting errors.
Left a minor comment.

**Review by Mykytako — APPROVED** [review]
Nice job!

**Review by suhrobmuboraksho — APPROVED** [review]
Comprehensive tests and docstring.
